module/execution/scripts.go

```diff
@@ -61,7 +61,7 @@ type Scripts struct {
 
 func NewScripts(
 	log zerolog.Logger,
-	metrics *metrics.ExecutionCollector,
+	metrics module.ExecutionMetrics,
```

Comment:
Is this related?

Reply:
it was for purpose of testing, but context wise no, so if you think it's better I can make a separate PR.